### ISSUE TITLE
Expected different QueryBuilder for relationMappings#modify

### ISSUE THREAD

**Issue by fl0w** [Vincit/objection.js#313]
So I'm wondering if I'm way off here or if I'm simply not in tune with inner workings of objection yet.
When defining a relation in relationMappings, I expected builder passed to `modify` to be instance of TargetModel's TargetQueryBuilder.

Consider following case (ES7/2016 for simplicity):

```js
class TargetQueryBuilder extends QueryBuilder {}

class TargetModel extends Model {
  static QueryBuilder = TargetQueryBuilder
  static RelatedQueryBuilder = TargetQueryBuilder
}

class MyModel extends Model {
  static relationMappings = {
    targetRelation: {
      relation: Model.HasManyRelation,
      modelClass: 'TargetModel',
      modify: builder => {
        console.log(builder.constructor.name)
        // -> MyModel#QueryBuilder
        // I expected builder to be an instance of TargetQueryBuilder
      },
      join: { ... }
    }
  }
}
```

I wanted use TargetQueryBuilder methods (in my specific case an inner join + some extra values, but unrelated to my question).

Is this expected behaviour or have I stumbled upon a use case off objection's scope? Perhaps I'm doing something wrong here or simply misusing modify/filter?

**Edit** Using `objection@0.7.0-rc.3`, `node. v7.6`

**Comment by fl0w** [comment]
Just to quickly add to this, `builder.modelClass()` within `#modify` is expected class, TargetModel.

**Comment by koskimas** [comment]
This seems like a bug. I'll check this out.

**Comment by koskimas** [comment]
Actually this is expected at the moment, but confusing and stupid as hell. This is how the query builders are used:

* `Model.QueryBuilder`: for `$query()` and `query()`
* `Model.RelatedQueryBuilder`: for `$relatedQuery()`

When you fetch `targetRelation` for `MyModel` something like this is done under the hood:

```js
myModelInstance.$relatedQuery('targetRelation').then(...)
```

and therefore the `MyModel`'s `RelatedQueryBuilder` is used.

This makes sense if you set the custom query builder for a base class so that every model uses the same query builder. If models have different query builders, this is crap.

I see three options here:

 1. fix objection: Remove `RelatedQueryBuilder` from Model. I don't know how useful it is. I have never used it.
 2. fix objection: Use `Model.QueryBuilder` in `eager`. This can be just as confusing if someone is taking advantage of the `RelatedQueryBuilder` feature.
3. Live with it.

What do you think?

**Comment by elhigu** [comment]
I would vote for simplicity, option 1. 

If later on someone needs different builder for related queries then we may reconsider and evaluate again if it is really necessary feature.

**Comment by fl0w** [comment]
Is it technically possible to shorthand a

```js
// use RelatedModel.QueryBuilder
myModelInstance
  .$relatedQuery('targetRelation')
  .useRelatedBuilder()
  .then(...)
// or perhaps
myModelInstance
  .$relatedQuery('targetRelation', true /* use target model QueryBuilder */)
  .then(...)
```

**Comment by koskimas** [comment]
This will be fixed with #343

**Comment by fl0w** [comment]
Alright, then I'll close this one.
